utils/getAbi.ts

```diff
@@ -51,6 +51,29 @@ export const Contracts: ContractType[] = [
     versions: ["2", "2.0.0", "2.0.1", "latest"],
     types: ["Perp", "Perpetual"],
     abi: perp2.abi,
+    async getState(instance: Contract) {
+      return {
+        collateralCurrency: (await instance.collateralCurrency()) as string, // address
+        priceIdentifier: (await instance.priceIdentifier()) as Bytes,
```

Comment:
The only place I could think of is if it needs to check the oracle price on settlement. In that case, it may need to pass in the price identifier as bytes, but we can always convert back. IMO, we should offer the most commonly used format.

Reply:
ive refactored the state to add `priceIdentifierUtf8` as well as the original `priceIdentifier` to stay as a buffer. Removed all places i could find where buffer was being converted to string and they now use `priceIdentifierUtf8` on the contract state instead.